README.md

```diff
@@ -148,28 +145,85 @@ One may specify multiple data locations by a list of locations as long as they h
+A path pointing to the Delta table location or a catalog path (when running on Databricks) of the format "catalog.schema.table", for more details on the table format, see [here](https://docs.databricks.com/data-governance/unity-catalog/index.html)
+<u>Examples</u>:
+  ```
+  location: catalog.schema.table
   ```
-  loader_method: load_file_as_dataframe
+  ```
+  location: "dbfs:/user/hive/warehouse/ml.db/all_prices_1990_2021_mar"
   ```
 
-- `sql`: string. Required if format is `spark_sql`.  
-Specifies a SparkSQL statement that identifies the dataset to use.
+- `version`: int. Optional.  
+Use this to specify the Delta table version to read from.
 
 
-- `version`: int. Optional.  
-If the `delta` format is specified, use this to specify the Delta table version to read from.
+- `timestamp`: timestamp. Optional.
+Use this to specify the timestamp at which to read data.
+</details>
+
+<details>
+<summary><strong><u>Using: "spark_sql"</u></strong></summary>
+
+- `location`: string. Optional.
+Dataset locations or the catalog path. If location is specified then automatically all the rows and columns are ingested without any selection. Either location or sql must be specified
+<u>Examples</u>:
+  ```
+  location: catalog.schema.table
+  ```
+  ```
+  location: "dbfs:/user/hive/warehouse/ml.db/all_prices_1990_2021_mar"
+  ```
 
+- `sql`: string. Optional.  
+Specifies a SparkSQL statement that identifies the dataset to use. Either location or sql must be specified
+<u>Examples</u>:
+  ```
+  sql: SELECT * FROM delta.`dbfs:/databricks-datasets/nyctaxi-with-zipcodes/subsampled`
+  ```
+
+</details>
+
+<details>
+<summary><strong><u>Using: "custom"</u></strong></summary>
 
-- `timestamp`: timestamp. Optional.  
-If the `delta` format is specified, use this to specify the timestamp at which to read data.
+- `location`: string. Required.  
+Dataset locations on the local filesystem are supported, as 
+well as HTTP(S) URLs and any other remote locations [resolvable by MLflow](https://mlflow.org/docs/latest/tracking.html#artifact-stores).
+One may specify multiple data locations by a list of locations as long as they have the same data format (see example below)
+<u>Examples</u>:
+  ```
+  location: ./data/sample.csv
+  ```
+  ```
+  location: https://raw.githubusercontent.com/fivethirtyeight/uber-tlc-foil-response/master/uber-trip-data/uber-raw-data-apr14.csv
+  ```
+  ```
+  location: ["./data/sample.parquet", "https://raw.githubusercontent.com/fivethirtyeight/uber-tlc-foil-response/master/uber-trip-data/uber-raw-data-apr14.csv"]
+  ```
+
+- `loader_method`: string.  
+Method name of the custom loader function from `steps/ingest.py`. The custom loader function allows use of datasets in other formats, such as `csv`. 
+The function should be defined in [`steps/ingest.py`](https://github.com/mlflow/recipes-regression-template/blob/main/steps/ingest.py),
+and should accept two parameters:
+- `file_path`: `str`. Path to the dataset file.
```

Comment:
Why do we need both `location` and `file_path` here? I think we only need the `loader_method` and `location`.